required/latex-lab/latex-lab-footnotes.dtx

```diff
@@ -495,39 +496,59 @@
 %    \item[\socket{tagsupport/fnmark} (1 argument)]
 %
 %      \DescribeSocket[noprint]{tagsupport/fnmark}
-%
+% 
+%    The socket is used in \cs{@footnotemark}/\cs{fnote_footnotemark:} 
+%    and takes \cs{@makefnmark} as argument. It prints the mark in the text
+%    and surrounds it with a tagging structure and a link. As such it is 
+%    not solely for tagging and so should not be used with \cs{UseTaggingSocket}
+%    as this would swallow the argument and loose the link support. 
+%    
 %      \fmi{describe and decide on names}
 %
 %
 %    \item[\socket{tagsupport/fntext/begin} (no argument)]
 %
 %      \DescribeSocket[noprint]{tagsupport/fntext/begin}
 %
+%    This socket is used before the main processing socket 
+%    (so before the \cs{insert} command). It opens the FEnote structure.
+%    As it sets also the tl-var for the current structure and this is used
+%    in destinations it should not use as tagging socket.
+%    
 %
 %    \item[\socket{tagsupport/fntext/end} (no argument)]
 %
 %      \DescribeSocket[noprint]{tagsupport/fntext/end}
 %
+%    This socket is used after the main processing socket 
+%    (so after the \cs{insert} command). It closes the FEnote structure.
 %
 %    \item[\socket{tagsupport/fntext/mark} (1 argument)]
 %
 %      \DescribeSocket[noprint]{tagsupport/fntext/mark}
-%
+%   
+%    This socket is used around the mark in the footnote text.   
+%    It adds tagging support but also link support, so like the other
+%    tagsupport sockets it should be always active.
 %
 %    \item[\socket{tagsupport/fntext/text} (1 argument)]
 %
 %      \DescribeSocket[noprint]{tagsupport/fntext/text}
 %
+%    This socket handles mc-chunks around the text of the footnote. As it
+%    takes an argument (the text) is should not be use as tagging socket either.
```

Comment:
I'll stop here, because I think this is a fundamental question that needs to be resolved first